spec/tictactoe_spec.rb

```diff
@@ -1,26 +1,123 @@
 require 'spec_helper' 
 require 'tictactoe' 
+## Faker Classes 
+class FakeIO
+  attr_accessor :stdin, :stdout 
+  def initialize(stdin: StringIO.new('') ) 
+      @stdin = stdin 
+      @stdout = []  
+  end 
+
+  def puts( string ) 
+    @stdout << string     
+  end 
+
+  def gets 
+    @stdin.gets @stdin.string 
+  end 
+end 
+
+def isRowOutputed( io: , row:  ) 
+    expect( io.stdout.include? row).to eql true 
+end 
+
 describe TicTacToe do 
-  context 'when the board is viewed' do
+
+  context 'when the board is rendered' do
     it 'returns a single list with 9 elements' do 
-     game = TicTacToe.new
+     game = TicTacToe.new(io: FakeIO.new)
 
      expect(game.renderBoard.length).to eql 9
-    end 
-
-    it 'returns a list of all strings' do 
-     game = TicTacToe.new
+     expect(game.renderBoard.all?{ |space| space.class == String && space.length == 1 } ).to eql true 
```

Comment:
What does this expectation mean? Is there something we could to do to reveal the intention a little better?